sprockets/mixins/amqp/mixins.py

```diff
@@ -0,0 +1,97 @@
+"""The PublishingMixin wraps RabbitMQ use into a request handler, with
+methods to speed the development of publishing RabbitMQ messages.
+
+Configured using two environment variables: ``AMQP_URL`` and ``AMQP_TIMEOUT``
+
+``AMQP_URL`` is the AMQP url to connect to, defaults to
+``amqp://guest:guest@localhost:5672/%2f``.
+
+``AMQP_TIMEOUT`` is the number of seconds to wait until timing out when
+connecting to RabbitMQ.
+
+"""
+import os
+import logging
+
+from tornado import gen
+
+from .amqp import AMQP
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PublishingMixin(object):
+    """This mixin adds publishing messages to RabbitMQ. It uses a
+    persistent connection and channel opened when the application
+    start up and automatically reopened if closed by RabbitMQ
+
+    """
+    @property
+    def app_id(self):
+        """Return a value to be used for the app_id AMQP message property.
+
+        :rtype: str
+
+        """
+        service = self.application.settings.get('service')
+        version = self.application.settings.get('version')
+        return self.generate_app_id(service, version)
+
+    @staticmethod
+    def generate_app_id(service, version):
+        return '{}/{}'.format(service, version)
+
+    @gen.coroutine
+    def publish_amqp_message(self,
+                             exchange,
+                             routing_key,
+                             body,
+                             properties=None,
+                             mandatory=False):
+        """Publish the message to RabbitMQ
+
+        Expects Correlation ID to be available via self.correlation_id,
+        default value will be None if not set.
+        The ``sprockets.mixins.correlation`` plugin provides this.
+
+        :param str exchange: The exchange to publish the message to
+        :param str routing_key: The routing key to publish the message with
+        :param bytes body: The message body to send
+        :param dict properties: An optional dict of additional properties
+                                to append. Will not override mandatory
+                                properties:
+                                app_id, correlation_id, message_id, timestamp
+        :param bool mandatory: Whether to instruct the server to return an
+                               unqueueable message
+                               http://www.rabbitmq.com/amqp-0-9-1-reference.html#basic.publish.mandatory
+
+        """
+        yield self.application.amqp.publish(
+            exchange,
+            routing_key,
+            body,
+            self.app_id,
+            getattr(self, 'correlation_id'),
+            properties,
+            mandatory
+        )
+
+
+def install(application, **kwargs):
```

Comment:
I wonder if this should this be in ``sprockets/mixins/amqp/__init__.py`` instead, seems a bit awkward for it to be in the mixin module.